### PR TITLE
chore: refactor schema

### PR DESCRIPTION
**Pull request by hoffa**
### Issue #, if available

#1133 

### Description of changes

Puts each resource in its own file. Also fixed some issues with `Api` schema (union at wrong granularity).

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [unit tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions) using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Do these changes include any template validations?
    - [ ] Did the newly validated properties support intrinsics prior to adding the validations? (If unsure, please review [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html) before proceeding).
        - [ ] Does the pull request ensure that intrinsics remain functional with the new validations?

### Examples?

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by GavinZZ — APPROVED** [review]
Wow nice! I love the refactor! Much cleaner to find each resource's schema.

File: .pylintrc [hunk 1]
```diff
@@ -108,6 +108,7 @@ disable=
     C0302, # too-many-lines
     W9013, # missing-yield-doc
     W9014, # missing-yield-type-doc
+    R0801, # duplicate-code, cannot be individually disabled (https://github.com/PyCQA/pylint/issues/214), enable when pylint updated
```
Comment (aahung) [inline .pylintrc:111]: we are already using the latest pylint and the issue seems to be closed

Comment (hoffa) [reply .pylintrc]: Aaaa good call, I tried a number of ways to put the disable comment but didn't work, finally figured out it has to be under the `class`...

Comment (aahung) [reply .pylintrc]: 😮‍💨

**Review by aahung — APPROVED** [review]

**Pull request merged**